resources/js/svir_viewer_d3_tree.js

```diff
@@ -256,24 +289,29 @@
 
                 // Get the values of the spinners
                 for (var i = 0; i < pdTempSpinnerIds.length; i++) {
-                    pdTempWeights.push($('#'+pdTempSpinnerIds[i]).val());
+                    var isInverted = $('#inverter-' + pdTempSpinnerIds[i]).is(':checked');
+                    var spinnerValue = $('#'+pdTempSpinnerIds[i]).val();
+                    if (isInverted) {
+                        spinnerValue = -spinnerValue;
+                    }
+                    pdTempWeights.push(spinnerValue);
                 }
 
                 // Adjust the values into percentages
                 pdTempWeights = pdTempWeights.map(Number);
                 var totalWeights = 0;
                 $.each(pdTempWeights,function() {
-                    totalWeights += this;
+                    totalWeights += Math.abs(this);
                 });
 
                 for (var i = 0; i < pdTempWeights.length; i++) {
-                    var tempMath = Math.floor((pdTempWeights[i] * 100) / totalWeights);
+                    var tempMath = (pdTempWeights[i] * 100) / totalWeights;
                     pdTempWeightsComputed.push(tempMath / 100);
                 }
 
                 // Update the results back into the spinners and to the d3.js chart
                 for (var i = 0; i < pdTempSpinnerIds.length; i++) {
```

Comment:
Also my editor complains, but it sounds reasonable to re-define a counter, instead of using a different name in each loop.